pkg/apis/serving/k8s_validation.go

```diff
@@ -586,6 +630,32 @@ func validateContainerPorts(ports []corev1.ContainerPort) *apis.FieldError {
 	return errs
 }
 
+func validateInitContainerPorts(ports []corev1.ContainerPort) *apis.FieldError {
+	var errs *apis.FieldError
+	if len(ports) == 0 {
+		return nil
+	}
+	for _, p := range ports {
+		errs = errs.Also(validateContainerPortBasic(p))
+	}
+	return errs
+}
```

Comment:
It looks like init containers ports are being treated differently than sidecar ports

we're skipping the checks
- only TCP is allowed
- only allow port names `http` `h2c` and `` (blank)


Thus I'm wondering what motivated the relaxed requirement and whether we should consider revisiting what we do for sidecars

Reply:
My pov is that init containers are normal app operation. So I thought users should be more flexible to use more stuff than normal Service operation allows.

Reply:
I agree - but let's pull this thread into an issue and see if anyone else has any opinions & thoughts.